### PR TITLE
Fix cli protocol parser

### PR DESCRIPTION
**Pull request by Jimbo4350**
#1451

### CODE REVIEW

**Comment by newhoggy** [comment]
Seems to not fix the problem for me in that `--epoch-slots` is still duplicated:

```
➜  cardano-node git:(jordan/fix-cli-protocol-parser) cabal build cardano-cli
Up to date
➜  cardano-node git:(jordan/fix-cli-protocol-parser) cabal exec -- cardano-cli shelley query tip
Usage: cardano-cli shelley query tip [--shelley-mode | --byron-mode
                                       [--epoch-slots NATURAL]
                                       [--security-param NATURAL] |
                                       --cardano-mode --epoch-slots NATURAL
                                       [--security-param NATURAL]]
                                     (--mainnet | --testnet-magic NATURAL)
                                     [--out-file FILE]
  Get the node's current tip (slot no, hash, block no)

Available options:
  --shelley-mode           For talking to a node running in Shelley-only mode.
  --byron-mode             For talking to a node running in Byron-only mode.
  --epoch-slots NATURAL    The number of slots per epoch for the Byron
                           era. (default: 21600)
  --security-param NATURAL The security parameter. (default: 2160)
  --cardano-mode           For talking to a node running in full Cardano mode
                           (default).
  --epoch-slots NATURAL    The number of slots per epoch for the Cardano era.
  --security-param NATURAL The security parameter. (default: 2160)
  --mainnet                Use the mainnet magic id.
  --testnet-magic NATURAL  Specify a testnet magic id.
  --out-file FILE          Optional output file. Default is to write to stdout.
```

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 1]
```diff
@@ -1951,6 +1951,15 @@ pProtocol =
           <> Opt.showDefault
           )
 
+    pEpochSlotsCardano :: Parser EpochSlots
+    pEpochSlotsCardano =
+      EpochSlots <$>
+        Opt.option Opt.auto
+          (  Opt.long "epoch-slots"
+          <> Opt.metavar "NATURAL"
+          <> Opt.help "The number of slots per epoch for the Cardano era."
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1960]: There isn't such a thing as the "Cardano" era, but either Byron or Shelley. The `EpochSlots` value only applies to the Byron era, so I think that this was correct the way it was before (using `pEpochSlots`).

**Review by intricate — CHANGES_REQUESTED** [review]
It doesn't seem that this addresses https://github.com/input-output-hk/cardano-node/issues/1451.

Also, I think that these particular changes (renaming `pEpochSlots` and adding `pEpochSlotsCardano`) should be reverted. See https://github.com/input-output-hk/cardano-node/pull/1827#discussion_r486004244.

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:1960]: So the conclusion would be to revert the changes and close the issue?

**Comment by Jimbo4350** [comment]
Closing, will revisit in the future

**Pull request closed without merging**